holidays/countries/tonga.py

```diff
@@ -0,0 +1,237 @@
+      - http://www.paclii.org/to/legis/num_act/pha2010243/  # Act 10 of 2010.
+      - http://www.paclii.org/to/legis/num_act/pha2013243/  # Act 5 of 2013.
+
+    Checked With:
+      - https://www.gov.to/press-release/tonga-public-holidays-for-2024/  # 2024
+      - https://www.gov.to/press-release/tonga-public-holidays-for-2023/  # 2023
+      - https://www.gov.to/press-release/tonga-public-holidays-for-2022/  # 2022
+      - https://www.gov.to/press-release/tonga-public-holidays-for-2021/  # 2021
+      - https://www.gov.to/press-release/tonga-public-holidays-for-2020/  # 2020
+      - https://www.gov.to/press-release/tonga-public-holidays-for-2019-2/  # 2019
+      - https://www.gov.to/press-release/tonga-public-holidays-for-2018/  # 2018
+      - https://www.officeholidays.com/countries/tonga/2017  # 2017
+
+    1988 Revision Observance Rule:
+        Provided always that when any of the days specified falls upon a Sunday,
+        the next following Monday shall be a public holiday and that whenever
+        the twenty-sixth day of December falls upon a Monday the day following
+        shall be a public holiday. (Amended by Act 11 of 1970.)
+
+    2016 and 2020 Revision Observance Rule:
+        Provided that when any public holidays specified, except Christmas Day, the
+        day immediately succeeding Christmas Day, New Years Day, Good Friday,
+        Easter Monday, ANZAC Day, Birthday of the reigning Sovereign of Tonga
+        and Birthday of the Heir to the Crown of Tonga, falls upon a Thursday,
+        Friday, Saturday or Sunday, that public holiday shall be celebrated on the next
+        following Monday; and if it falls on a Tuesday or Wednesday, that public
+        holiday shall be celebrated on the Monday before the actual public holiday.
+        (Amended by Act 10 of 2010: May 18, 2010.)
+        "Birthday of the reigning Sovereign of Tonga and Birthday of the Heir to
+        the Crown of Tonga" is add to the exempted list as seen above.
+        (Amended by Act 5 of 2013: Jun 28, 2013.)
+
+        Further provided that the Birthday of the reigning Sovereign of Tonga and the
+        Birthday of the Heir to the Crown of Tonga shall be celebrated on the day it
+        falls, unless it falls on a Sunday in which case it would be celebrated on the
+        next following Monday.
+        (Inserted by Act 5 of 2013: Jun 28, 2013.)
+    """
+
+    country = "TO"
+    default_language = "to"
+    # %s (Observed).
+    observed_label = tr("%s (Fakatokanga'i)")
+    supported_languages = ("en_US", "to")
+
+    def __init__(self, *args, **kwargs):
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        StaticHolidays.__init__(self, cls=TongaStaticHolidays)
+        kwargs.setdefault("observed_rule", SUN_TO_NEXT_MON)
+        super().__init__(*args, **kwargs)
+
```

Comment:
```suggestion

    def _add_observed(self, dt: date, **kwargs):
        if self._year >= 2010 and kwargs.get("rule") is None:
            kwargs["rule"] = TUE_WED_TO_PREV_MON + THU_FRI_TO_NEXT_MON + SAT_SUN_TO_NEXT_MON
        return super()._add_observed(dt, **kwargs)

```

Reply:
That's exactly what I'm looking for, tysm for the help <3